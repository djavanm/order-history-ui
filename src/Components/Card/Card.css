section {
  padding: 20px;
}

img {
  height: 200px;
  width: 200px;
}

.purchase-card {
  background: white;
  box-shadow: 0px 0px 16px 0px rgba(0,0,0,0.75);
  padding: 10px;
  display: flex;
  justify-content: space-evenly;
  margin: 20px;
}

.purchase-info {
  display: flex;
  flex-direction: column;
  width: 45%;
  text-align: left;
}

h3 {
  font-size: 1.5em;
  margin: 10px;
}

p {
  font-size: 1em;
  margin-top: 40px;
}