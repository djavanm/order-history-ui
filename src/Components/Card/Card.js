import React from 'react';
import './Card.css';

const Card = ({ purchase }) => {
  const { name, img, price, description } = purchase;
  return (
    <article className="purchase-card">
      <div className="purchase-image">
      <img src={img} alt={`${name} that was bought`} />
      </div>

      <div className="purchase-info">
      <h3>{name}</h3>
      <p>Description: {description}</p>
      </div>

      <div className="purchase-price">
      <h3>Price {price}</h3>
      </div>

    </article>
  )
}

export default Card;