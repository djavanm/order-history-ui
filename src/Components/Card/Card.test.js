import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card' , () => {
  let mockData;
  beforeEach(() => {
    mockData = {
      id: 1,
      name: 'Djavan',
      price: 9999, 
      description: 'Great for the whole family',
      img: 'random corner of the internet'
    }
  })
  it('should match the snapshot with the mock data', () => {
    const wrapper = shallow(<Card purchase={mockData}
    />)
    expect(wrapper).toMatchSnapshot();
  })
});