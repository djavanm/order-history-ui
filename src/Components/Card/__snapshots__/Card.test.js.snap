// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with the mock data 1`] = `
<article
  className="purchase-card"
>
  <div
    className="purchase-image"
  >
    <img
      alt="Djavan that was bought"
      src="random corner of the internet"
    />
  </div>
  <div
    className="purchase-info"
  >
    <h3>
      Djavan
    </h3>
    <p>
      Description: 
      Great for the whole family
    </p>
  </div>
  <div
    className="purchase-price"
  >
    <h3>
      Price 
      9999
    </h3>
  </div>
</article>
`;
